libr/anal/sign.c

```diff
@@ -1545,7 +1545,7 @@ static void print_list_type_body(RAnal *a, RSignItem *it, PJ *pj, int format) {
 		i++;
 	}
 
-	if (format == 'j') {
+	if (format == 'j' && i > 2) {
```

Comment:
uhm seems hacky. whats going on inside print_type_json that only works if there are two types printed first?

Reply:
It's in [`print_list_seperator`](https://github.com/radareorg/radare2/blob/0adbb4abec422ac5432d177d12232948bcf94eb2/libr/anal/sign.c#L1520)

I'm not sure why this can be an array in any case for JSON output. And apparently I think this is more broken. For example I have a function with 3 function args, while `z` shows them correctly, `zj` does not... I'll try to fix this and provide some test cases.